AP-protocol.md

```diff
@@ -157,141 +166,86 @@ Message is subject to fragmentation: see the dedicated section.
-		file_size: u64,
-		file: String,
-	},
-	ErrorFileNotFound, //S -> C : error_file_not_found!
-}
-
-enum MediaMessage{
-	MediaRequest(MediaRequest),
-	MediaResponse(MediaResponse)
-}
-
-struct MediaRequest{
-	Media{  //C -> S : media?(media_id, +media type+ )
-		media_id: u64
-		///media: media_kind
-	}
+#[derive(Debug)]
+pub struct MessageData { // Only part fragmentized
+	session_id: u64,
+	content: MessageContent
 }
+```
 
-enum MediaResponse{
-	MediaResponse { //S -> C : media!(media_size, media, +media type+)
-		media_id: u64,
-		media_size: u64,
-		media: std::fs::File,
-	},
-	ErrorNoMediaResponse,
-	ErrorMediaNotFoundResponse,//S -> C : error_media_not_found!
+#### Message Types
+```rust
+#[derive(Debug)]
+pub enum MessageContent{
+	// Client -> Server
+	ReqServerType,
+	ReqFilesList,
+	ReqFile(u64),
+	ReqMedia(u64),
+
+	ReqClientList,
+	ReqMessageSend { to: NodeId, message: Vec<u8> },
+	// Do we need request of new messages? or directly sent by server?
+
+	// Server -> Client
+	RespServerType(ServerType)
+	RespFilesList(Vec<u64>),
+	RespFile(Vec<u8>),
+	RespMedia(Vec<u8>),
+	ErrUnsupporedRequestType,
+	ErrRequestedNotFound
+
+	RespClientList(Vec<NodeId>),
+	RespMessageFrom { from: NodeId, message: Vec<u8> },
+	ErrWrongClientId,
 }
 ```
 
-### Error
-
-If a drone receives a Message and the next hop specified in the Source Routing Header is not a neighbor of the drone, then it sends Error to the client.
-
-This message cannot be dropped by drones due to Packet Drop Rate.
-
+Example of new request:
 ```rust
-struct Error {
-	session_id: u64,
-	/// ID of drone, server of client that is not a neighbor:
-	id_not_neighbor: String,
-	ttl: u32,
-}
+let routing = getRoutingHeader();
+let content = MessageType:ReqFile(8);
+Message::new(routing, source_id, session_id, content)
 ```
 
-Source Routing Header contains the path to the client, which can be obtained by reversing the list of hops contained in the Source Routing Header of the problematic Message.
 
-### Dropped
+### NACK
+If an error occurs that a NACK is sent. A NACK can be of type:
+1. **ErrorInRouting**: If a drone receives a Message and the next hop specified in the Source Routing Header is not a neighbor of the drone, then it sends Error to the client.
+2. **Dropped**: If a drone receives a Message that must be dropped due to the Packet Drop Probability, then it sends Dropped to the client.
 
-If a drone receives a Message that must be dropped due to the Packet Drop Rate, then it sends Dropped twice to the client.
+Source Routing Header contains the path to the client, which can be obtained by reversing the list of hops contained in the Source Routing Header of the problematic Message.
 
-This message cannot be dropped by drones due to Packet Drop Rate.
+This message cannot be dropped by drones due to Packet Drop Probability.
 
 ```rust
-struct Dropped {
-	session_id: u64,
+pub enum Nack{
+	ErrorInRouting(NodeId), // contains id of not neighbor
+	Dropped
```

Comment:
i think it would be useful to keep track of which fragment we have dropped, just to know which one we have to send again